adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,274 @@
+	}
+}
+
+func getReferer(request *openrtb2.BidRequest) string {
+	if request.Site == nil {
+		return ""
+	}
+
+	return request.Site.Domain
+}
+
+func getCurrency(request *openrtb2.BidRequest) string {
+	if len(request.Cur) == 0 {
+		return "USD"
+	}
+
+	return request.Cur[0]
+}
+
+func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var (
+		requests []*adapters.RequestData
+		errors   []error
+	)
+
+	referer := getReferer(requestData)
+	currency := getCurrency(requestData)
+
+	if referer == currency {
+		return nil, nil
+	}
+
+	for i := range requestData.Imp {
+		imp := requestData.Imp[i]
+		bidType, err := getBidType(imp)
+
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		splittedRequestData := processDataFromRequest(requestData, imp, bidType)
+
+		requestBody, err := json.Marshal(splittedRequestData)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requests = append(requests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpointUri,
+			Body:    requestBody,
+			Headers: getHeaders(requestData),
+		})
+	}
+
+	return requests, errors
+}
+
+func processDataFromRequest(requestData *openrtb2.BidRequest, imp openrtb2.Imp, bidType openrtb_ext.BidType) resetDigitalRequest {
+
+	var resetDigitalRequestData resetDigitalRequest
+	resetDigitalRequestData.Site.Domain = requestData.Site.Domain
+	resetDigitalRequestData.Site.Referrer = requestData.Site.Page
+
+	resetDigitalRequestData.Imps = append(resetDigitalRequestData.Imps, resetDigitalRequesImps{})
+	resetDigitalRequestData.Imps[0].BidID = requestData.ID
+	resetDigitalRequestData.Imps[0].ImpID = imp.ID
+
+	var err error
+
+	if bidType == openrtb_ext.BidTypeBanner {
+		resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{imp.Banner.Format[0].W, imp.Banner.Format[0].H})
+	}
+	if bidType == openrtb_ext.BidTypeVideo {
+		resetDigitalRequestData.Imps[0].MediaTypes.Video.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{*imp.Video.W, *imp.Video.H})
+	}
+
+	var extData = make(map[string]interface{})
+	err = json.Unmarshal(imp.Ext, &extData)
+	if err != nil {
+
+	} else {
+
+		resetDigitalRequestData.Imps[0].ZoneID.PlacementID = extData["bidder"].(map[string]interface{})["placement_id"].(string)
+		if resetDigitalRequestData.Imps[0].ZoneID.PlacementID == "test" {
+			resetDigitalRequestData.Imps[0].ForceBid = true
+		}
+
+	}
+	return resetDigitalRequestData
+
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, nil
+	}
+	if response.StatusCode != http.StatusOK {
+		return nil, nil
+	}
```

Comment:
could make use of response util here

https://github.com/prebid/prebid-server/blob/ec729e6e5d50a1306d76332fda52108b9d313c1d/adapters/response.go#L10-L28